### PR TITLE
Feature ranking

### LINKED ISSUE #24: Feature show Ranking

### PR DESCRIPTION
**Pull request by PedroCarrillo**
Close #24

### CODE REVIEW

File: app/src/main/java/com/belatrixsf/allstars/adapters/MainNavigationViewPagerAdapter.java [hunk 1]
```diff
@@ -44,7 +47,7 @@ public Fragment getItem(int position) {
             case 0:
                 return AccountFragment.newInstance(null);
             case 1:
-                return AccountFragment.newInstance(null);
+                return new RankingContainerFragment();
```
Comment (cpinan) [inline app/src/main/java/com/belatrixsf/allstars/adapters/MainNavigationViewPagerAdapter.java]: I see we are using static in the others implementations. Why this one is just calling the constructor?

File: app/src/main/java/com/belatrixsf/allstars/ui/common/views/AccountSelectionView.java [hunk 2]
```diff
@@ -42,7 +42,7 @@ public void setData(String data) {
     }
 
     public void setProfileImage(String imageUrl) {
-        Glide.with(profileImageView.getContext()).load(imageUrl).override(50, 50).centerCrop().transform(new CircleTransform(profileImageView.getContext())).into(profileImageView);
+        Glide.with(profileImageView.getContext()).load(imageUrl).override(50, 50).centerCrop().transform(new BorderedCircleTransformation(profileImageView.getContext())).into(profileImageView);
```
Comment (cpinan) [inline app/src/main/java/com/belatrixsf/allstars/ui/common/views/AccountSelectionView.java]: This size should be in dimens.xml

File: app/src/main/java/com/belatrixsf/allstars/ui/ranking/RankingContainerFragment.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                } else if (position == TAB_ALL_TIME && !wasSelected) {
+                    replaceChildFragment(RankingFragment.newInstance(Constants.KIND_SCORE));
+                }
+            }
+        });
+    }
+
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
```
Comment (cpinan) [inline app/src/main/java/com/belatrixsf/allstars/ui/ranking/RankingContainerFragment.java]: I think we can remove this

File: app/src/main/res/drawable/ic_event.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+        android:width="24dp"
+        android:height="24dp"
+        android:viewportWidth="24.0"
+        android:viewportHeight="24.0">
+    <path
+        android:fillColor="#FF000000"
```
Comment (cpinan) [inline app/src/main/res/drawable/ic_event.xml:7]: This looks like a transparent color we can use @android:color/transparent, same for the size, 24dp should be in dimens.xml

File: app/src/main/res/drawable/ic_star.xml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+        android:width="24dp"
+        android:height="24dp"
+        android:viewportWidth="24.0"
+        android:viewportHeight="24.0">
```
Comment (cpinan) [inline app/src/main/res/drawable/ic_star.xml:5]: Same here as above

File: app/src/main/res/drawable/ic_whatshot.xml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+        android:width="24dp"
+        android:height="24dp"
+        android:viewportWidth="24.0"
+        android:viewportHeight="24.0">
+    <path
```
Comment (cpinan) [inline app/src/main/res/drawable/ic_whatshot.xml:6]: Same here

File: app/src/main/res/layout/fragment_account.xml [hunk 7]
```diff
@@ -16,51 +16,57 @@
...
+                            android:layout_marginTop="@dimen/dimen_3_5"
+                            tools:text="Pedro Carrillo"
+                            style="@android:style/TextAppearance.Large" />
+
+                        <TextView
+                            android:id="@+id/profile_role"
+                            android:layout_width="wrap_content"
+                            android:layout_height="wrap_content"
+                            android:layout_marginTop="@dimen/dimen_1_5"
+                            tools:text="Senior Android Developer" />
```
Comment (cpinan) [inline app/src/main/res/layout/fragment_account.xml:56]: This is unreal xD

File: app/src/main/res/layout/fragment_account.xml [hunk 8]
```diff
@@ -163,8 +169,8 @@
 
         <ImageView
             android:id="@+id/profile_picture"
-            android:layout_width="@dimen/dimen_10_8"
-            android:layout_height="@dimen/dimen_10_8"
+            android:layout_width="120dp"
+            android:layout_height="120dp"
```
Comment (cpinan) [inline app/src/main/res/layout/fragment_account.xml]: Replace by dimens

File: app/src/main/res/layout/fragment_ranking_container.xml [hunk 9]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.v4.widget.NestedScrollView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:tag="@string/ranking_tag"
+    android:paddingBottom="56dp"
```
Comment (cpinan) [inline app/src/main/res/layout/fragment_ranking_container.xml]: Dimens here

File: app/src/main/java/com/belatrixsf/allstars/adapters/RankingListAdapter.java [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+                    cupResourceId = R.drawable.ic_first_place_cup;
+                    break;
+                case 2:
+                    crownResourceId = R.drawable.ic_second_place;
+                    cupResourceId = R.drawable.ic_second_place_cup;
+                    break;
+                default:
+                    crownResourceId = R.drawable.ic_third_place;
+                    cupResourceId = R.drawable.ic_third_place_cup;
+                    break;
```
Comment (cpinan) [inline app/src/main/java/com/belatrixsf/allstars/adapters/RankingListAdapter.java:85]: I think this is not maintainable maybe we can create a static final array, something like:

public static final CROWN_RESOURCES[] = {..., ..., ...}
public static final CUP_RESOURCES[] = {..., ..., ...} 

and use crownResource = CROWN_RESOURCES[place-1] same for cups and put the constants in the Constants class

File: app/src/main/res/drawable/ic_event.xml (hunk 4 above)
Comment (PedroCarrillo) [inline app/src/main/res/drawable/ic_event.xml:7]: this is a vector :octocat:

File: app/src/main/res/drawable/ic_star.xml (hunk 5 above)
Comment (PedroCarrillo) [inline app/src/main/res/drawable/ic_star.xml:5]: this is a vector :octocat:

File: app/src/main/res/drawable/ic_whatshot.xml (hunk 6 above)
Comment (PedroCarrillo) [inline app/src/main/res/drawable/ic_whatshot.xml:6]: this is a vector :octocat:

File: app/src/main/res/layout/fragment_account.xml (hunk 7 above)
Comment (PedroCarrillo) [inline app/src/main/res/layout/fragment_account.xml:56]: HAHAHAHAHa

File: app/src/main/java/com/belatrixsf/allstars/adapters/RankingListAdapter.java (hunk 10 above)
Comment (PedroCarrillo) [inline app/src/main/java/com/belatrixsf/allstars/adapters/RankingListAdapter.java:85]: file an issue as an enhacemente for this.

**Pull request merged**